src/theory/quantifiers/sygus_inference.cpp

```diff
@@ -160,7 +169,7 @@ bool SygusInference::simplify(std::vector<Node>& assertions)
   }
 
   // if no free function symbols, there is no use changing into SyGuS
-  if (free_functions.empty())
+  if (free_functions_uf.empty())
```

Comment:
please add an option to represent this behavior of only inferring when non-constant free functions exist

Reply:
Actually i think the default behavior should be to always do this. I think if the user doesnt want sygus inference to kick in in this case, they should just disable the original option. I'll remove this check.